### PR TITLE
[GraphQL] Document identification specs.

### PR DESCRIPTION
**Pull request by alloy**
cc @ashkan18 @mzikherman

### CODE REVIEW

File: playbooks/graphql-schema-design.md [hunk 1]
```diff
@@ -123,6 +135,24 @@ e.g. an `artworks` field _and_ an `artworks_connection`, simply choose one form
...
+## Global Object Identification
+
+All GraphQL services should follow the [Global Object Identification specification](https://facebook.github.io/relay/graphql/objectidentification.htm). Due to schema stitching, for their IDs services should encode their own service ID for metaphysics to be able to resolve a node ID back to its upstream service.
+
+For instance, Exchange should encode an Order with ID 42 as follows:
+
+```
+Base64("Exchange:Order:42")
+```
+
```
Comment (orta) [inline playbooks/graphql-schema-design.md]: Maybe recommend always lowercasing everything for consistency? (though maybe non-number IDs might fail on that?)

**Comment by alloy** [comment]
cc @l2succes

File: playbooks/graphql-schema-design.md [hunk 2]
```diff
@@ -123,6 +135,28 @@ e.g. an `artworks` field _and_ an `artworks_connection`, simply choose one form
 - In rare cases an immediate list may be used, but this should only be done in cases where the list has few entries
   and doesn’t require pagination, i.e. fetch _all_ entries at once.
 
+## ID fields
+
+An ID field that refers to e.g. a database ID should be called something like `internalID`, it MUST _never_ be called
```
Comment (mzikherman) [inline playbooks/graphql-schema-design.md]: Interesting! So does this mean there won't be the official support of specifying an `__id` as an `idField`? I thought (at least in MP) we adhere to that, so therefore continuing to support `id` as a database ID or slug was ok.

But, also `internalID` sounds totally fine to me as being prescribed, so wondering if I'm missing some active movement around next steps with the Relay fork.

File: playbooks/graphql-schema-design.md [hunk 3]
```diff
@@ -123,6 +135,28 @@ e.g. an `artworks` field _and_ an `artworks_connection`, simply choose one form
...
+just `id`, as that name is reserved for ‘Global Object Identification’ (further explained in the next section).
+
+Its type should be `ID!`, which is a custom string scalar meant to convey that the value is an identifier and is not
+nullable, as database IDs never are `null`.
+
+## Global Object Identification
+
+All GraphQL services should follow the [Global Object Identification specification](https://facebook.github.io/relay/graphql/objectidentification.htm). Due to schema stitching, for their IDs services should encode their own service ID for metaphysics to be able to resolve a node ID back to its upstream service.
```
Comment (mzikherman) [inline playbooks/graphql-schema-design.md]: Good call, esp with the Postgres primary keys, there would def be collisions (less common w/ the Mongo timestamped ones) so including a type and service is 👍

File: playbooks/graphql-schema-design.md (hunk 2 above)
Comment (alloy) [reply playbooks/graphql-schema-design.md]: We’ll be having a larger discussion in the platform practice after my vacation, but it does seem ideal to get off of the fork sooner rather than later–as we won’t know how long any official support, if any, may still take.

**Comment by alloy** [comment]
Even though this does seem to be where we’re headed, perhaps we shouldn’t merge this as-is yet until we’ve come to a conclusion within the platform practice on the direction.

File: playbooks/graphql-schema-design.md (hunk 2 above)
Comment (zephraph) [reply playbooks/graphql-schema-design.md]: @mzikherman some graphql implementations now give a [deprecation warning](https://github.com/graphql/graphql-js/blob/77c36183ff46d7d8853f8d6fe1051ebd59d89392/src/utilities/assertValidName.js#L37) if you explicitly express fields with `__`. It seems those are reserved for internal usage and code relying on it may break in future releases. 

Beyond that, the global object identification spec requires that the `id` field be globally unique (and subsequently traceable to its origin). 

Basically just means that how metaphysics is currently running requires that we maintain our relay fork (and potentially forks of graphql server implementations if they start reserving `__` in a breaking way). To get away from that we have to both change how `id` is used and remove usage of `__id`.

File: playbooks/graphql-schema-design.md (hunk 1 above)
Comment (zephraph) [reply playbooks/graphql-schema-design.md]: 👍 for system names being lower cased for consistency. Type names should likely exactly reflect the type name in the schema and I don't think we should do anything to the `id`.

**Comment by zephraph** [comment]
Hey @alloy maybe it's time to merge?

**Comment by alloy** [comment]
Let’s do it 👍
